libnavigation-base/src/main/java/com/mapbox/navigation/base/utils/route/RouteUtils.kt

```diff
@@ -0,0 +1,163 @@
+package com.mapbox.navigation.base.utils.route
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.geojson.utils.PolylineUtils
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.utils.DecodeUtils.precision
+import com.mapbox.navigation.utils.internal.logE
+import com.mapbox.navigation.utils.internal.logW
+
+/**
+ * Route Utils
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+object RouteUtils {
+
+    private const val LOG_CATEGORY = "RouteUtils"
+
+    /**
+     * The method checks whenever the route has unexpected closures based
+     * on [RouteOptions.snappingIncludeClosuresList] and [RouteOptions.snappingIncludeStaticClosuresList]
+     * flags. If the any of them is **false** for the waypoint and closure overlaps it the method
+     * immediately returns **true**.
+     *
+     * If route has closures that are not overlapped waypoints, the method immediately returns
+     * **true**.
+     *
+     * Note: Experimental preview API
+     */
+    @ExperimentalPreviewMapboxNavigationAPI
+    fun hasRouteUnexpectedClosures(navigationRoute: NavigationRoute): Boolean {
+        val snappingResultList = navigationRoute.directionsRoute.getSnappingResultList()
+
+        var index = -1
+        navigationRoute.directionsRoute.legs()?.forEachIndexed { routeLegIndex, routeLeg ->
+            index++
+
+            val legLastGeometryIndex = routeLeg.stepsGeometryToPoints(
+                navigationRoute.directionsRoute.precision()
+            ).lastIndex
+
+            val silentWaypoints = routeLeg.silentWaypoints()
+
+            routeLeg.closures()?.forEach { closure ->
+                val silentWaypointsInClosureRange = silentWaypoints.inGeometryRange(
+                    closure.geometryIndexStart(), closure.geometryIndexEnd()
+                )
+
+                if (closure.geometryIndexStart() == 0 &&
+                    snappingResultList.getOrNull(index) != true
+                ) {
+                    logW(
+                        "Route with id [${navigationRoute.id}] has closure at the start of the" +
+                            " leg, leg index $routeLegIndex",
+                        LOG_CATEGORY
+                    )
```

Comment:
No need for a log, this is an expected result of the function. Same below.

Reply:
let's keep it with `logD` to have more info in case of any issues